### ISSUE TITLE
yum module fails when package already installed and latest version

### ISSUE THREAD

**Issue by ttbigc** [ansible/ansible-modules-core#3177]
##### Issue Type:
- Bug Report
##### Ansible Version: 2.0.1

ansible 2.0.0.1
##### Ansible Configuration:

default
##### Environment:

Ansible Master: Oracle Linux 7 
Remote Hosts: Oracle Linux 5.11
##### Summary:

```
tasks:
  yum: name={{ item }} state=latest
  with_items:
   "{{ ntp_package_list }}"
```

Initial run of playbook works without issue however subsequent runs I get the following error:
TASK [ntp : Install via YUM] ***************************************************
Wednesday 02 March 2016  16:17:53 -0600 (0:00:00.199)       0:00:50.351 *******
failed: [XXXXXXXXXX] => (item=[u'ntp']) => {"failed": true, "item": ["ntp"], "module_stderr": "", "module_stdout": "{\"msg\": \"\", \"invocation\": {\"module_args\": {\"name\": [\"ntp\"], \"list\": null, \"install_repoquery\": true, \"conf_file\": null, \"disable_gpg_check\": false, \"state\": \"latest\", \"disablerepo\": null, \"update_cache\": false, \"enablerepo\": null, \"exclude\": null}}, \"changed\": false, \"results\": [\"All packages providing ntp are up to date\", \"\"], \"rc\": 0}\r\nException exceptions.AttributeError: \"'NoneType' object has no attribute 'unshare_data'\" in <bound method YumBase.__del__ of <yum.YumBase object at 0x4e3f290>> ignored\r\n", "msg": "MODULE FAILURE", "parsed": false}

**Comment by ttbigc** [comment]
Seems to be issue with repo/config as after making some changes I am no longer encountering this issue. Closing issue at this time as I believe env specific issue. If I run into again will try to debug the yum module to find cause.

**Comment by azureliang** [comment]
I met a similar error, how did you modify your repo/config?

failed: [10.252.155.73] => (item=[u'sudo']) => {"failed": true, "invocation": {"module_name": "yum"}, "item": ["sudo"], "module_stderr": "OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_request_forwards: requesting forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 295\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to 10.252.155.73 closed.\r\n", "module_stdout": "{\"msg\": \"\", \"invocation\": {\"module_args\": {\"name\": [\"sudo\"], \"list\": null, \"install_repoquery\": true, \"conf_file\": null, \"disable_gpg_check\": false, \"state\": \"installed\", \"disablerepo\": null, \"update_cache\": false, \"enablerepo\": null, \"exclude\": null}}, \"changed\": true, \"results\": [\"Loaded plugins: security\nExcluding Packages in global exclude list\nFinished\nSetting up Install Process\nResolving Dependencies\n--> Running transaction check\n---> Package sudo.x86_64 0:1.7.2p1-29.el5_10 set to be updated\n--> Finished Dependency Resolution\n\nDependencies Resolved\n\n================================================================================\n Package      Arch           Version                   Repository          Size\n================================================================================\nInstalling:\n sudo         x86_64         1.7.2p1-29.el5_10         el5_latest         364 k\n\nTransaction Summary\n================================================================================\nInstall       1 Package(s)\nUpgrade       0 Package(s)\n\nTotal download size: 364 k\nDownloading Packages:\nRunning rpm_check_debug\nRunning Transaction Test\nFinished Transaction Test\nTransaction Test Succeeded\nRunning Transaction\n\r  Installing     : sudo                                                     1/1 \nwarning: /etc/sudoers created as /etc/sudoers.rpmnew\n\nInstalled:\n  sudo.x86_64 0:1.7.2p1-29.el5_10                                               \n\nComplete!\n\"], \"rc\": 0}\r\nException exceptions.AttributeError: \"'NoneType' object has no attribute 'unshare_data'\" in <bound method YumBase.__del__ of <yum.YumBase object at 0x1d77d10>> ignored\r\n", "msg": "MODULE FAILURE", "parsed": false}

**Comment by ttbigc** [comment]
Guess I should of updated my findings when I closed this issue.
I found that Oracle released an update to yum RPM (yum-3.2.22-40.0.2.el5) for Oracle Linux 5 that breaks ansible during certain yum_module tasks (mainly when RPM already installed).
To resolve I just excluded this package and used these tasks to correct:
- name: Check if yum-3.2.22-40.0.2.el5 installed
  shell: rpm -q yum-3.2.22-40.0.2.el5
  register: yum
  changed_when: False
  failed_when: False
- name: Downgrade yum-3.2.22-40.0.2 to yum-3.2.22-40.0.1
  shell: rpm -Uvh --oldpackage http://{{ yum_server }}/Oracle/Linux/5Server/x86_64/current-latest/yum-3.2.22-40.0.1.el5.noarch.rpm
  when: yum.rc == 0

You could also patch ansible to handle the different behavior of the Oracle modified yum version but I felt it was better to downgrade yum as the Oracle modification seemed irrelevant.

**Comment by azureliang** [comment]
Thanks a lot! according to some policy, I can not downgrade the yum in the image, so maybe I can only do somthing on  ansible
